packages/botonic-cli/src/commands/deploy.ts

```diff
@@ -345,9 +340,27 @@ Uploading...
       track('Deploy Botonic Provider Error', { error: e })
       console.log(colors.red(`There was an error getting the providers: ${e}`))
     }
+  }
+
+  async deploy() {
     try {
-      await exec('rm botonic_bundle.zip')
-    } catch (e) {}
-    this.botonicApiService.beforeExit()
+      let build_out = await this.botonicApiService.buildIfChanged(
+        force,
+        npmCommand
+      )
+      if (!build_out) {
+        track('Deploy Botonic Build Error')
+        console.log(colors.red('There was a problem building the bot'))
+        return
+      }
+      await this.createBundle()
```

Comment:
It's okai, but I think we should start using this `await Promise.all([someCall(), anotherCall()])` when we have more than one async function.

Reply:
mmm.. but in this way some methods can be executed not in the right order, right?
I mean, by calling Promise.all, we allow deployBundle to be run before createBundle, for example, or am I wrong?